### PR TITLE
Improve error message when assign wrong attributes to model

### PR DESCRIPTION
**Pull request by artofhuman**
### Summary

Include class name in `ArgumentError` message for a more precise understanding of the error

### CODE REVIEW

**Comment by artofhuman** [comment]
@rafaelfranca Can u review this change?

File: activemodel/lib/active_model/attribute_assignment.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@ module AttributeAssignment
     #   cat.status # => 'sleeping'
     def assign_attributes(new_attributes)
       if !new_attributes.respond_to?(:stringify_keys)
-        raise ArgumentError, "When assigning attributes, you must pass a hash as an argument."
+        raise ArgumentError, "When assigning attributes, you must pass a hash as an argument, #{new_attributes.class} passed."
```
Comment (gmcgibbon) [inline activemodel/lib/active_model/attribute_assignment.rb]: I wonder if we should say "hash-like object" instead of simply "hash"?

**Review by gmcgibbon — APPROVED** [review]
Still relevant in `6.0.0.alpha`. Seems reasonable to me. This behaviour will cascade to `#update` / `#update_attributes` for active record objects too.

File: activemodel/lib/active_model/attribute_assignment.rb (hunk 1 above)
Comment (eileencodes) [reply activemodel/lib/active_model/attribute_assignment.rb]: I think "hash-like object" would be confusing to beginners. Hash is fine.

**Review by eileencodes — COMMENTED** [review]
Can you remove the changelog entry? It's not necessary since it's not a feature or a major change in behavior.

**Comment by artofhuman** [comment]
@eileencodes @gmcgibbon Thanks for the review. I removed changelog entry.

File: activemodel/test/cases/attribute_assignment_test.rb [hunk 2]
```diff
@@ -95,9 +95,11 @@ def dup
   end
 
   test "an ArgumentError is raised if a non-hash-like object is passed" do
-    assert_raises(ArgumentError) do
+    err = assert_raises(ArgumentError) do
       Model.new(1)
     end
+
+    assert_equal("When assigning attributes, you must pass a hash as an argument, #{1.class} passed.", err.message)
```
Comment (eugeneius) [inline activemodel/test/cases/attribute_assignment_test.rb]: `1.class` was only necessary for compatibility with Ruby 2.3; master now requires Ruby 2.4 or newer, so you can just use `Integer` here.

**Pull request merged**